docs/execution-pipeline.md

```diff
@@ -1,59 +1,59 @@
-During execution phase, we perform a traversal over execution plan tree, create an execution context objects for each of the fields and supply them along with data resolved from previous higher level resolver responses recrusively.
+During execution phase, we perform a traversal of the execution plan tree; create execution context objects for each of the fields and supply them along with data resolved from previous higher level resolver responses recursively.
 
-Field resolution context contains all of the necessary metadata about things such as:
+A field resolution context contains metadata such as:
 
-- Executing schema itself.
-- Variables provided with the request.
-- Field parameters (in case if field defined any arguments).
-- Type definition of both parent and expected return type.
-- ExecutionInfo which is fragment of the execution plan itself related to current field and all of its subsequent fields.
+- Information about the execution of the schema itself;
+- Variables provided with the request;
+- Field parameters (in case the corresponding field defined any arguments);
+- Type definition of both parent and expected return type;
+- `ExecutionInfo` which is a fragment of the execution plan itself related to the current field and all of its subsequent fields.
 
-Execution phase can be performed using one of the two strategies:
+The execution phase can be performed using one of the two strategies:
 
-- **Parallel**, where asynchronous fields can be resolved in unordered manner. This is default option in case of GraphQL queries due to fact, that they are readonly operations.
-- **Sequential** where asynchronous fields must be resolved in correct order (as it was defined in query string). This is default mode for GraphQL mutations.
+- **Parallel**: in which asynchronous fields can be resolved in any order. This is the default mode for GraphQL queries due to the fact that they are readonly operations.
+- **Sequential**: in which asynchronous fields must be resolved in a specific order: as it was defined in the query. This is default mode for GraphQL mutations.
 
-As result of GraphQL query execution, a dictionary with resolved values will be returned. It contains following fields:
+The result of a GraphQL query execution is a dictionary with resolved values. This dictionary contains the following fields:
 
-- `documentId` which is query AST document hash code - it can be used to implement execution plan caching.
-- `data` with a formated GraphQL response matching the requested query string.
-- `errors` entry is optional. If it has been provided, it will contain list of errors, that occured during query execution.
+- `documentId`: which is the hash code of the query's AST document - it can be used to implement execution plan caching (persistent queries).
+- `data`: with a formated GraphQL response matching the requested query.
+- `errors`: optional. If it has been provided, it will contain a list of errors that occured during query execution.
```

Comment:
```suggestion
- `documentId`: which is the hash code of the query's AST document - it can be used to implement execution plan caching (persistent queries).
- `data`: optional, a formatted GraphQL response matching the requested query (`KeyValuePair seq`). Absent in case of an error that does not allow continuing processing and returning any GraphQL results.
- `errors`: optional, contains a list of errors (`GQLProblemDetails`) that occurred during query execution.
```